src/Token.php

```diff
@@ -232,6 +232,9 @@ public function getClaim($name, $default = null)
         }
 
         if ($name === RegisteredClaims::AUDIENCE && is_array($value)) {
+            if (count($value) > 1) {
+                trigger_error('Behaviour for audience claims has changed in 3.4, you will only get the first array entry as a string. In 4.0 this will change again with more flexibility, please consult the documentation.', E_USER_DEPRECATED);
```

Comment:
The behaviour didn't change. The library was never designed to have an array as value for the audience - you can see that on the doc block of all methods (even on the old interface), this `if` just made that explicit.

Using `withClaim()` or `setClaim()` to configure `aud` worked back then because things were overlooked and I've assumed people would use the specific methods to define registered claims.

Given we already have a deprecation message to prevent that, I'm not convinced we need yet another deprecation message here.

Reply:
This deprecation message is targeting the readback from a token, and if some external system is adding more than one audiences as an array, this will trigger on accessing the claim.